public/pages/workflow_detail/workspace/component_details.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useContext } from 'react';
+import { useOnSelectionChange } from 'reactflow';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiPanel,
+  EuiTitle,
+  EuiEmptyPrompt,
+  EuiText,
+} from '@elastic/eui';
+import { ReactFlowComponent } from '../../../../common';
+import { rfContext } from '../../../store';
+import { InputFieldList } from '../workspace_component/input_field_list';
+
+// styling
+import './workspace-styles.scss';
+
+interface ComponentDetailsProps {
+  onToggleChange: () => void;
+  isOpen: boolean;
+}
+
+/**
+ * A panel that will be nested in a resizable container to dynamically show
+ * the details and user-required inputs based on the selected component
+ * in the flow workspace.
+ */
+export function ComponentDetails(props: ComponentDetailsProps) {
+  // TODO: use this instance to update the internal node state. ex: update field data in the selected node based
+  // on user input
+  const { reactFlowInstance } = useContext(rfContext);
+
+  const [selectedComponent, setSelectedComponent] = useState<
+    ReactFlowComponent
+  >();
+
+  /**
+   * Hook provided by reactflow to listen on when nodes are selected / de-selected.
+   * - populate panel content appropriately
+   * - open the panel if a node is selected and the panel is closed
+   */
+  useOnSelectionChange({
+    onChange: ({ nodes, edges }) => {
+      if (nodes && nodes.length > 0) {
+        setSelectedComponent(nodes[0]);
```

Comment:
why is this hard coded to `nodes[0]`?

Reply:
It is only possible to select one node at once as far as I'm aware - there are certain dynamic grouping options available with reactflow pro which may mean the array would be >1 but I'm unable to find any examples from the docs or any way to trigger >1 node selected when testing things. It may be for future-proofing if added functionality in the free or Pro versions open up multiple selections.

Reply:
Ah I see, didn't know there was a pro version. Seemed weird if its an array that we can't grab multiple and then would need some sort of filter function to only set one selected component. maybe add a small comment?

Reply:
Will add